homeassistant/components/growatt_server/sensor.py

```diff
@@ -8,345 +8,30 @@
 import voluptuous as vol
 
 from homeassistant.components.sensor import PLATFORM_SCHEMA, SensorEntity
-from homeassistant.const import (
-    CONF_NAME,
-    CONF_PASSWORD,
-    CONF_USERNAME,
-    CURRENCY_EURO,
-    DEVICE_CLASS_BATTERY,
-    DEVICE_CLASS_CURRENT,
-    DEVICE_CLASS_ENERGY,
-    DEVICE_CLASS_POWER,
-    DEVICE_CLASS_TEMPERATURE,
-    DEVICE_CLASS_VOLTAGE,
-    ELECTRICAL_CURRENT_AMPERE,
-    ENERGY_KILO_WATT_HOUR,
-    FREQUENCY_HERTZ,
-    PERCENTAGE,
-    POWER_WATT,
-    TEMP_CELSIUS,
-    VOLT,
-)
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME
 import homeassistant.helpers.config_validation as cv
 from homeassistant.util import Throttle
 
+from .growatt_sensor_types import (
```

Comment:
Please do the refactor in a separate PR than this PR which is a feature addition.

Reply:
You happy for me to do that as an amend & force push over the top of this one?

Reply:
It's ok to amend this time, since I haven't really looked in detail. Normally we don't want commits to be squashed after review has started.

Reply:
Yeah understood, that's my normal policy too.
I prefer to squash as much as possible up until the point there's comments on specific lines of code.

Reply:
Updated as requested and force-pushed over the top of the original submission.
The only thing I think may cause some contention is the fact I've changed the update period to 1 minute instead of 5. 
The GrowattServer is updated every 5 minutes but if you only update HASS every 5 minutes as well there can be a massive lag between the stats.
It seems excessive to change the update period in it's own PR but if that's what it takes, just let me know.